lib/SILGen/SILGenExpr.cpp

```diff
@@ -4283,6 +4284,12 @@ RValue RValueEmitter::visitProtocolMetatypeToObjectExpr(
   return RValue(SGF, E, v);
 }
 
+RValue RValueEmitter::visitFromUninhabitedExpr(FromUninhabitedExpr *E,
+                                               SGFContext C) {
+  SGF.emitIgnoredExpr(E->getSubExpr());
```

Comment:
Let's just emit the undef - evaluating the conversion's operand doesn't make sense if the program is going to terminate.

Reply:
Don't we need to evaluate the expression here to ensure that the program does indeed terminate, especially if the expression has side effects? For example, in let x = y ?? fatalError("message"), the subExpr will be the call to fatalError, which we want to emit code for. I might be misunderstanding SILGen for uninhabited types, should I be emitting an unreachable instruction before the undef?

Reply:
Aha. Makes sense - the operand is guaranteed to have an unreachable (or divergent code pattern) somewhere in its calling sequences so this is fine.

Retracted.